packages/@lwc/template-compiler/src/parser/attribute.ts

```diff
@@ -211,24 +211,20 @@ function isInputStateAttribute(element: IRElement, attrName: string) {
 }
 
 export function isAttribute(element: IRElement, attrName: string): boolean {
-    const isCustom = isCustomElement(element);
-    if (isCustom) {
+    const isCustomElm = isCustomElement(element);
+    if (isCustomElm) {
         return isCustomElementAttribute(attrName);
     }
 
-    // Handle global attrs (common to all tags) and special attribute (role, aria, key, is, data-).
-    if (isGlobalHtmlAttribute(attrName) || isAriaOrDataOrFmkAttribute(attrName)) {
-        return true;
-    }
-
     // Handle input tag value="" and checked attributes that are only used for state initialization.
     // Because .setAttribute() won't update the value, those attributes should be considered as props.
     if (isInputStateAttribute(element, attrName)) {
```

Comment:
I would recommend inline the function here. The comment above explains what the `isInputStateAttribute` doing. Moving the content in place would help the overall code readability.

```suggestion
    if (element.tag === 'input' && (attrName === 'value' || attrName === 'checked')) {
```